components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/internal/ClientManager.java

```diff
@@ -0,0 +1,142 @@
+
+import static java.util.Objects.isNull;
+
+/**
+ * Class to retrieve the HTTP Clients.
+ */
+public class ClientManager {
+
+    private static final Log LOG = LogFactory.getLog(ClientManager.class);
+    private final CloseableHttpAsyncClient httpAsyncClient;
+
+    /**
+     * Creates a client manager.
+     *
+     * @throws WebSubAdapterException on errors while creating the http client.
+     */
+    public ClientManager() throws WebSubAdapterException {
+
+        LOG.info("Initializing ClientManager");
+        PoolingNHttpClientConnectionManager connectionManager;
+        try {
+            connectionManager = createPoolingConnectionManager();
+            LOG.info("Successfully created PoolingNHttpClientConnectionManager");
+        } catch (IOException e) {
+            throw WebSubHubAdapterUtil.handleServerException(WebSubHubAdapterConstants.ErrorMessages.ERROR_CREATING_ASYNC_HTTP_CLIENT, e);
+        }
+
+        RequestConfig config = createRequestConfig();
+        HttpAsyncClientBuilder httpClientBuilder = HttpAsyncClients.custom().setDefaultRequestConfig(config);
+        addSslContext(httpClientBuilder);
+        httpClientBuilder.setConnectionManager(connectionManager);
+        httpAsyncClient = httpClientBuilder.build();
+        httpAsyncClient.start();
+        LOG.info("HttpAsyncClient started");
+    }
+
+    /**
+     * Get HTTP client properly configured with tenant configurations.
+     *
+     * @return CloseableHttpAsyncClient instance.
+     */
+    public CloseableHttpAsyncClient getClient() throws WebSubAdapterException {
+
+        if (isNull(httpAsyncClient)) {
+            LOG.error("HttpAsyncClient is null");
+            throw WebSubHubAdapterUtil.handleServerException(WebSubHubAdapterConstants.ErrorMessages.ERROR_GETTING_ASYNC_CLIENT, null);
+        } else if (!httpAsyncClient.isRunning()) {
+            LOG.warn("HttpAsyncClient is not running, starting client");
+            httpAsyncClient.start();
+        }
+        return httpAsyncClient;
+    }
+
+    private RequestConfig createRequestConfig() {
+
+        return RequestConfig.custom()
+                .setConnectTimeout(WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                        .getHTTPConnectionTimeout())
+                .setConnectionRequestTimeout(WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                        .getHttpConnectionRequestTimeout())
+                .setSocketTimeout(
+                        WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().getHttpReadTimeout())
+                .setRedirectsEnabled(false)
+                .setRelativeRedirectsAllowed(false)
+                .build();
+    }
+
+    private PoolingNHttpClientConnectionManager createPoolingConnectionManager() throws IOException {
+
+        int maxConnections = WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                .getDefaultMaxConnections();
```

Comment:
what's the default ?

Reply:
20 - https://github.com/wso2-enterprise/asgardeo-product/blob/master/repository/conf/identity/identity-outbound-adapter.properties#L16